modules/client/src/addressList/internal/client/methods.ts

```diff
@@ -275,9 +277,17 @@ export class ClientAddressListMethods extends ClientCore
       if (!addressListProposal) {
         return null;
       }
-      // format in the metadata field
-      const metadataString = await this.ipfs.fetchString(addressListProposal.metadata);
-      // TODO: Parse and validate schema
+      let metadataCid = "";
+      try {
+        metadataCid = resolveIpfsCid(addressListProposal.metadata);
+        // TODO: Parse and validate schema
+      } catch (err) {
```

Comment:
We have another try/catch block outside, why not reusing it and detecting the error type?

Reply:
Because we need to do the return and we dont have `dao`, `proposalAddressList` and `proposalErc20` inside the the outer catch